mantis-common/src/main/java/io/mantisrx/runtime/descriptor/SchedulingInfo.java

```diff
@@ -135,27 +180,73 @@ public Builder multiWorkerScalableStageWithConstraints(int numberOfWorkers, Mach
                                                                StageScalingPolicy scalingPolicy) {
             StageScalingPolicy ssp = new StageScalingPolicy(currentStage, scalingPolicy.getMin(), scalingPolicy.getMax(),
                     scalingPolicy.getIncrement(), scalingPolicy.getDecrement(), scalingPolicy.getCoolDownSecs(), scalingPolicy.getStrategies());
-            builderStages.put(currentStage, new StageSchedulingInfo(numberOfWorkers, machineDefinition,
-                    hardConstraints, softConstraints, ssp, ssp.isEnabled()));
+            builderStages.put(
+                currentStage,
+                StageSchedulingInfo.builder()
+                    .numberOfInstances(numberOfWorkers)
+                    .machineDefinition(machineDefinition)
+                    .hardConstraints(hardConstraints)
+                    .softConstraints(softConstraints)
+                    .scalingPolicy(ssp)
+                    .scalable(ssp.isEnabled())
+                    .build());
             currentStage++;
             return this;
         }
 
         public Builder multiWorkerStageWithConstraints(int numberOfWorkers, MachineDefinition machineDefinition,
                                                        List<JobConstraints> hardConstraints, List<JobConstraints> softConstraints) {
-            builderStages.put(currentStage, new StageSchedulingInfo(numberOfWorkers, machineDefinition,
-                    hardConstraints, softConstraints, null, false));
+            builderStages.put(
+                    currentStage,
+                    StageSchedulingInfo.builder()
+                            .numberOfInstances(numberOfWorkers)
+                            .machineDefinition(machineDefinition)
+                            .hardConstraints(hardConstraints)
+                            .softConstraints(softConstraints)
+                            .build());
             currentStage++;
             return this;
         }
 
         public Builder multiWorkerStage(int numberOfWorkers, MachineDefinition machineDefinition) {
-            builderStages.put(currentStage, new StageSchedulingInfo(numberOfWorkers, machineDefinition,
-                    null, null, null, false));
+            return multiWorkerStage(numberOfWorkers, machineDefinition, false);
+        }
+
+        public Builder multiWorkerStage(int numberOfWorkers, MachineDefinition machineDefinition, boolean scalable) {
+            builderStages.put(
+                    currentStage,
+                    StageSchedulingInfo.builder()
+                            .numberOfInstances(numberOfWorkers)
+                            .machineDefinition(machineDefinition)
+                            .scalable(scalable)
+                            .build());
             currentStage++;
             return this;
         }
 
+        public Builder createWithInstanceInheritance(
```

Comment:
The scheduling builder is a bit counter-intuitive because it has to set the number of stages separately from "add*" calls. When using this method writer needs to call the set stage number again to adjust the stage count. I will add java doc on this one.